test/encryption/encryption.test.js

```diff
@@ -185,6 +184,90 @@ describe('encryption integration tests', () => {
       });
     }
 
+    describe('mongoose Maps', function() {
+      describe('CSFLE', function() {
+        it('encrypts and decrypts', async function() {
+          const schema = new Schema({
+            a: {
+              type: Schema.Types.Map,
+              of: String,
+              encrypt: { keyId: [keyId], algorithm }
+
+            }
+          }, {
+            encryptionType: 'csfle'
+          });
+
+          connection = createConnection();
+          const model = connection.model('Schema', schema);
+          await connection.openUri(process.env.MONGOOSE_TEST_URI, {
+            dbName: 'db', autoEncryption: {
+              keyVaultNamespace: 'keyvault.datakeys',
+              kmsProviders: { local: { key: LOCAL_KEY } },
+              extraOptions: {
+                cryptdSharedLibRequired: true,
+                cryptSharedLibPath: process.env.CRYPT_SHARED_LIB_PATH
+              }
+            }
+          });
+
+          const [{ _id }] = await model.insertMany([{ a: {
+            name: 'bailey'
+          } }]);
+          const encryptedDoc = await utilClient.db('db').collection('schemas').findOne({ _id });
+
+          assert.ok(isBsonType(encryptedDoc.a, 'Binary'));
+          assert.ok(encryptedDoc.a.sub_type === 6);
+
+          const doc = await model.findOne({ _id });
+          assert.ok(doc.a instanceof Map);
+          const rawObject = Object.fromEntries(doc.a);
+          assert.deepEqual(rawObject, { name: 'bailey' });
+        });
+      });
+
+      describe('queryable encryption', function() {
+        it('encrypts and decrypts', async function() {
+          const schema = new Schema({
+            a: {
+              type: Schema.Types.Map,
+              of: String,
+              encrypt: { keyId: keyId }
+
+            }
+          }, {
+            encryptionType: 'queryableEncryption'
+          });
+
+          connection = createConnection();
+          const model = connection.model('Schema', schema);
+          await connection.openUri(process.env.MONGOOSE_TEST_URI, {
+            dbName: 'db', autoEncryption: {
+              keyVaultNamespace: 'keyvault.datakeys',
+              kmsProviders: { local: { key: LOCAL_KEY } },
+              extraOptions: {
+                cryptdSharedLibRequired: true,
+                cryptSharedLibPath: process.env.CRYPT_SHARED_LIB_PATH
+              }
+            }
+          });
+
+          const [{ _id }] = await model.insertMany([{ a: {
+            name: 'bailey'
+          } }]);
+          const encryptedDoc = await utilClient.db('db').collection('schemas').findOne({ _id });
+
+          assert.ok(isBsonType(encryptedDoc.a, 'Binary'));
+          assert.ok(encryptedDoc.a.sub_type === 6);
+
+          const doc = await model.findOne({ _id });
```

Comment:
Minor question: can you actually query by encrypted map fields using queryable encryption?

Reply:
No - queries are limited when using QE.  Here's the documentation on supported query types: https://www.mongodb.com/docs/manual/core/queryable-encryption/reference/supported-operations/#supported-and-unsupported-bson-types